packages/table-core/src/core/column.ts

```diff
@@ -52,12 +52,16 @@ export function createColumn<TData extends RowData, TValue>(
         let result = originalRow as Record<string, any>
 
         for (const key of accessorKey.split('.')) {
-          result = result[key]
-          if (process.env.NODE_ENV !== 'production' && result === undefined) {
-            throw new Error(
-              `"${key}" in deeply nested key "${accessorKey}" returned undefined.`
-            )
+          if (result == null || !(key in result)) {
+            if (process.env.NODE_ENV !== 'production') {
+              console.warn(
+                `"${key}" in deeply nested key "${accessorKey}" returned undefined.`
+              )
+            }
+            return null
           }
+
+          result = result[key]
```

Comment:
shouldn't this still be above the if statement?

Reply:
The result assignment is moved to the last line because you cannot read properties of null/undefined (which is covered by the if statement above it).